remember.go

```diff
@@ -0,0 +1,97 @@
+package remember
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+type Memory struct {
+	input []io.Reader
+	output io.ReadWriter
+}
+
+type Option func(*Memory) error
+
+func WithInput(args []string) Option {
+	return func(m *Memory) error {
+		var input = []io.Reader{}
+		for _, str := range args {
+			input = append(input, bytes.NewBufferString(str))
+		}
+		m.input = input
+		return nil
+	}
+}
+
+func WithOutput(output string) Option {
+	return func(m *Memory) error {
+		m.output = bytes.NewBufferString(output)
+		return nil
+	}
+}
+
+func NewMemory(opts ...Option) (Memory, error) {
+	_, err := os.Stat("store.txt")
+	if err != nil {
+		os.Create("store.txt")
+	}
+	defaultFile, err := os.OpenFile("store.txt", os.O_APPEND|os.O_RDWR, 0644)
+	m := Memory{
+		[]io.Reader{os.Stdin},
+		defaultFile,
+	}
+	for _, opt := range opts {
+		err := opt(&m)
+		if err != nil {
+			return Memory{}, err
+		}
+	}
+	return m, nil
+}
+
+func (m *Memory) Recall() string {
+	data, err := ioutil.ReadAll(m.output)
```

Comment:
Using ReadAll will allocate memory to hold the entire file being read. This is alright if you know input will not be large. IF you are concerned about a user exhausting memory by providing a large input file, you can use `io.Copy()` to move the data in chunks between input and output `io.Reader`s.

Reply:
That is a useful thing to know, I will keep it in mind in the future. However for this use case I am happy too offload the responsibility to the user on this one for now. 

Having said that, it does raise some interesting questions about input validation, ie what if a user mistakenly passed a binary fill and filled their datastore with junk